File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandResponseMessages/GraphBuildingEdgeCreationSuccessResponseMsg.java
package InfoSecCooker.Communication.Messages.CommandResponseMessages;

public class GraphBuildingEdgeCreationSuccessResponseMsg extends GraphBuildingSuccessResponseMsg
{
    private long createdEdgeId;
    private long createdEdgeTemporaryId;

    public GraphBuildingEdgeCreationSuccessResponseMsg(long createdEdgeId, long createdEdgeTemporaryId, String additionalInfo)
    {
        super(additionalInfo);
        this.createdEdgeId = createdEdgeId;
        this.createdEdgeTemporaryId = createdEdgeTemporaryId;
    }

    public GraphBuildingEdgeCreationSuccessResponseMsg(long createdEdgeId, long createdEdgeTemporaryId)
    {
        super("");
        this.createdEdgeId = createdEdgeId;
        this.createdEdgeTemporaryId = createdEdgeTemporaryId;
    }
}

File: JavaScript Front End/src/Messages/CommandResponseMessages/GraphBuildingEdgeCreationSuccessResponseMsg.js
const GraphBuildingSuccessResponseMsg = require('./GraphBuildingSuccessResponseMsg.js').GraphBuildingSuccessResponseMsg;

class GraphBuildingEdgeCreationSuccessResponseMsg extends GraphBuildingSuccessResponseMsg
{
    constructor(jsonMessage)
    {
        super(jsonMessage);
        this.createdEdgeId = this.jsonMessage["createdEdgeId"];
        this.createdEdgeTemporaryId = this.jsonMessage["createdEdgeTemporaryId"];
    }
}

module.exports = {GraphBuildingEdgeCreationSuccessResponseMsg};
